lib/elixir/lib/exception.ex

```diff
@@ -765,6 +765,24 @@ defmodule UnicodeConversionError do
   end
 end
 
+defmodule StartLinkNameError do
+  defexception [arg: nil]
+
+  def message(exception) do
+    """
+    invalid argument type given for `name`.
+
+    Valid types are:
+      * nil
+      * atom
+      * tuple
```

Comment:
These docs may be helpful: http://elixir-lang.org/docs/master/elixir/GenServer.html#module-name-registration